### ISSUE TITLE
Loads tiles outside viewport after rotation

### ISSUE THREAD

**Issue by KevinBeckers** [openseadragon/openseadragon#1289]
Hi,

When I rotate the image the following happens:
- Tiles outside the view are loaded.
- Tiles inside the view aren't ( Keeps black )

I guess there is something wrong with the center position. Or the rotation point.
In the image below you can see that the left stays black. The top and right are loaded even if this is outside the viewport.

![ezgif-2-1e9b44caa5](https://user-images.githubusercontent.com/4037137/29612280-065baa36-8801-11e7-92a2-5a6111a221a3.gif)

How can I fix this? Is this a known issue?

**Comment by KevinBeckers** [comment]
The rotation point is the problem.
It looks like the viewportMargins causes this.

**Comment by iangilman** [comment]
@KevinBeckers Is this rotating the viewport or rotating an individual image? And it only happens when you have viewport margins on? 

@avandecreme Are you up for taking a look at this?

**Comment by KevinBeckers** [comment]
@iangilman Yes this happens only with the margins. This is to correct the image. By default, it is not centered. For you information I do the rotation with: viewer.viewport.setRotation(rotation);

**Comment by iangilman** [comment]
Okay, good to know. If you'd like to dig in to the issue, this line might be a good place to start: 

https://github.com/openseadragon/openseadragon/blob/master/src/tiledimage.js#L994

You could try just changing that from `getBoundsWithMargins` to `getBounds`. 

That said, I'm curious about your use of margins. In what way is the image not centered? There are potentially other ways to solve that issue than using margins.

**Comment by KevinBeckers** [comment]
I will try to take a look. This week I will have some time.
Thanks for the start.

I feel like the black margin around the image is not even. So I used margin to correct this. How do you suggest to fix this?

**Comment by iangilman** [comment]
The black margin around the image should be even by default! Can you share a repro of that issue? 

Thank you for looking into the rotation issue. :)

**Comment by avandecreme** [comment]
Isn't this a duplicate of #1007 ?

**Comment by iangilman** [comment]
@avandecreme Good point! I guess I'll keep them both around since they both have discussion; thank you for linking them though.

**Comment by iangilman** [comment]
@KevinBeckers The discussion in #1007 might be useful as you pursue the issue...
